frontend/app/Settings/RefreshButton.js

```diff
@@ -1,21 +1,46 @@
 'use client';
 
 import CachedIcon from '@mui/icons-material/Cached';
-import { useCallback, useContext } from 'react';
+import { useCallback, useContext, useEffect } from 'react';
 import useQueryParams from '../../lib/hooks/useQueryParams';
 
 import styles from './RefreshButton.module.scss';
 import classNames from 'classnames/bind';
 import { ViewContext } from '../contexts/ViewContext';
+import { ConfigContext } from '../contexts/ConfigContext';
 const cx = classNames.bind(styles);
 
+
 const RefreshButton = ({ query }) => {
     const { params, updateParams } = useQueryParams();
     const { columns, view } = useContext(ViewContext);
+    const { config } = useContext(ConfigContext);
+
+    const checkForUpdates = useCallback(async () => {
+        if (!!query?.timestamp) {
+            try {
+                fetch(`${config.rootUrl}api/timestamp?dgid=${query?.dgid}`).then(async res => {
+                    const currentTimestamp = (await res.json()).timestamp;
+                    if (!!currentTimestamp) {
+                        if (query.timestamp !== currentTimestamp)
+                            updateParams({...params});
+                    }
+                });
+            } catch (error) {
+                console.log(error);
+            }
+        }
+    }, [query?.timestamp, query?.dgid, params, updateParams]);
 
+    useEffect(async () => {
```

Comment:
I don't think this is achieving what you want. I believe the useEffect hook does not support async functions in this manner (they'd need to be called as promises within the actual effect function), and the interval variable would need to be defined outside of the effect hook (probably as a ref). Also, because the dependency array is empty, this hook won't be firing except for on initial render.